### PR TITLE
Use ChatCompletion (gpt-3.5-turbo). Include state via sqlite

### PR DESCRIPTION
**Pull request by fxchen**
This PR does two things:
- uses gpt-3.5-turbo (alternatively gpt-4) for each openai call.
- uses a sqlite database to be able to continue from the last task.

image

### CODE REVIEW

File: babyagi.py [hunk 1]
```diff
@@ -18,6 +20,36 @@
 print("\033[96m\033[1m"+"\n*****OBJECTIVE*****\n"+"\033[0m\033[0m")
 print(OBJECTIVE)
 
+# Initialize SQLite database
+db_conn = sqlite3.connect("miniagi.db")
+db_cursor = db_conn.cursor()
+
+db_cursor.execute("""
+CREATE TABLE IF NOT EXISTS tasks (
```
Comment (davidbyttow) [inline babyagi.py:28]: problem with this approach is that there's only one task list with no way to reset, right?

should probably have a namespace/identifier that is given (via args or env or whatever) such that one can build a new task list from scratch or continue from a previous one without having to manually delete their DB

File: babyagi.py [hunk 2]
```diff
@@ -120,6 +185,14 @@ def context_agent(query: str, index: str, n: int):
         print("\033[93m\033[1m"+"\n*****TASK RESULT*****\n"+"\033[0m\033[0m")
         print(result)
 
+        # Store the result in the task_results and completed_tasks table
+        db_cursor.execute(
```
Comment (davidbyttow) [inline babyagi.py:189]: code is fine but i think next step is to abstract this out into a task store that uses sqlite (or whatever other pluggable means such as json files)

**Review by davidbyttow — COMMENTED** [review]
fyi

File: babyagi.py (hunk 1 above)
Comment (davidbyttow) [reply babyagi.py:28]: i guess the best namespace to use would be the pinecone table name

**Comment by l4time** [comment]
Would have been better to have a PR for each. This project desperately needs to include gpt-3.5-turbo to reduce api cost.

**Comment by davidbyttow** [comment]
Agree @l4time: https://github.com/yoheinakajima/babyagi/pull/41

**Review by francip — COMMENTED** [review]
I'd split this into two separate PRs. Also, the SQLite storage probably can be put in a separate file and included based on environment variable to replace the current in-memory implementation, to keep the core as small and clean as @yoheinakajima wants.

**Comment by webbfly** [comment]
@l4time I agree. I am just wondering if I am missing some benefits davinci-003 has over gpt3.5 turbo, so that it is used in this project?

**Comment by fxchen** [comment]
@yoheinakajima @francip @davidbytow - let me know if there's appetite for external state storage in sqlite. I'll recreate this functionality.

Let's close this PR for now - the branch is based on the first or so master commit.

**Pull request closed without merging**